### PR TITLE
VBA Fields validation

### PR DESCRIPTION
**Pull request by akshayasalvi**
@puneetlath Can you please review this?

### Details
- Added regex for phone validation
- Added maxLength and keyboardType for fields

### Fixed Issues

$ https://github.com/Expensify/App/issues/5184

### Tests
1. Tested the phone number with brackets and with long digits
2. Tested ssn with more than 4 digits
3. Tested taxIDNumber with more than 9 digits
4. Tested zipCode with more than 5 digits

### QA Steps
There are two forms where the validation is added.

**Company Step**
1. Got to `bank-account/company-step`
2. Enter invalid values for `zipCode`, `taxIDNumber` and `phoneNumber`
3. Click on `Save & Continue`
4. Validate that fields show the error

**Requestor Step** 
1. Got to `bank-account/personal-information`
2. Enter invalid values for `ssn`
3. Click on `Save & Continue`
4. Validate that fields show the error

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web

web-ssn-error
web-ssn-valid
web-validation-errors
web-valid-values

#### Mobile Web

mweb-validation-errors
mweb-valid-values-001
mweb-valid-values-002
mweb-ssn-error
mweb-ssn-valid

#### Desktop

desktop-validation-errors
desktop-valid-values
desktop-ssn-error
desktop-ssn-valid

#### iOS

ios-validation-errors
ios-valid-values
ios-ssn-error
ios-ssn-valid

#### Android

android-validation-errors
android-valid-values
android-ssn-error
android-ssn-value

### CODE REVIEW

**Comment by akshayasalvi** [comment]
@puneetlath We might have to update the placeholder for `Phone number` field. At the moment it says `10 digits, no hyphens`. Can you or @SofiedeVreese help me with the new placeholder that I can update?

**Comment by Luke9389** [comment]
@akshayasalvi Regarding the phone number placeholder, I think it's fine to just use `Phone Number (xxx)xxx-xxxx` .

Also, I think you should use `Str.isValidPhone` from `expensify-common` in the `validate()` method on this form.

**Comment by Luke9389** [comment]
Hey @akshayasalvi, great job so far on this issue. 👍 

We are wanting to push through all issues related to this flow very quickly, so it's possible that I'll end up taking this PR and finishing it personally. Rest assured, you will still get paid as if you completed this issue. I'll check back in tomorrow and make a determination then.

**Comment by akshayasalvi** [comment]
> @akshayasalvi Regarding the phone number placeholder, I think it's fine to just use `Phone Number (xxx)xxx-xxxx` .

Same placeholder in Spanish too?

> Also, I think you should use `Str.isValidPhone` from `expensify-common` in the `validate()` method on this form.

I had seen that but it seems our current `isValidPhone` only allows digits and not parenthesis? I had even raised it in the discussion here https://github.com/Expensify/App/issues/5184#issuecomment-922929708


> Hey @akshayasalvi, great job so far on this issue. 👍

Thank you

> We are wanting to push through all issues related to this flow very quickly, so it's possible that I'll end up taking this PR and finishing it personally. Rest assured, you will still get paid as if you completed this issue. I'll check back in tomorrow and make a determination then.

Okay both the changes that you've mentioned have follow-up questions. Can you help so that I'll update the PR at the earliest?

**Comment by Luke9389** [comment]
> Same placeholder in Spanish too?

That's a good point. For Spanish, lets do `(prefijo) + (número)`.

**Comment by Luke9389** [comment]
> I had seen that but it seems our current isValidPhone only allows digits and not parenthesis? I had even raised it in the discussion here #5184 (comment)

You also offered taking out the parentheses in that comment, which sounds like a fine solution to me. We don't need to make that a separate method, just do it in one line.

File: src/CONST.js
Comment (Luke9389) [inline src/CONST.js:428]: Didn't see this originally, you can ignore my suggestion to use `Str.isValidPhone`

However, this doesn't limit the number of digits after the area code. I'm not sure if it's best to always rely on using `MAX_LENGTH` in combination with `PHONE_WITH_SPECIAL_CHARS`. Does that make sense?

File: src/libs/ValidationUtils.js
Comment (Luke9389) [inline src/libs/ValidationUtils.js:78]: This doesn't test for max length. I know we do that on the input, but  I worry that this will be used without that.

**Review by Luke9389 — COMMENTED** [review]
I think this is looking really good. Let me know what you think about my concern with de-coupling the max length of phone numbers and their symbols.

File: src/CONST.js
Comment (akshayasalvi) [reply src/CONST.js:428]: What is max length of the digits after the area code?

**Comment by akshayasalvi** [comment]
> I think this is looking really good. Let me know what you think about my concern with de-coupling the max length of phone numbers and their symbols.

@Luke9389 I've updated the regex with length check on the numbers. I've also added an additional max length check in the function itself.

Also, I saw two conflicts where I've picked my changes. I insist on keeping my change `keyboardType={CONST.KEYBOARD_TYPE.NUMERIC}` whereas the incoming change had it set as `KEYBOARD_TYPE.PHONE_PAD` with fields like `zipCode` and `ssnLast4` being numbers only.

File: src/CONST.js
Comment (Luke9389) [inline src/CONST.js:427]: I think this brings us to a max of 14 digits total, right? I think we want to bump this to `{0,12}`

File: src/CONST.js
Comment (Luke9389) [inline src/CONST.js]: I don't understand why this was moved out of `BANK_ACCOUNT.MAX_LENGTH`. Would you explain that for me?

**Review by Luke9389 — COMMENTED** [review]
Getting closer, just a few more questions. Thanks!

File: src/CONST.js
Comment (akshayasalvi) [reply src/CONST.js]: Sorry about this. Updated.

File: src/CONST.js
Comment (akshayasalvi) [reply src/CONST.js]: I thought phone number being a generic field we might use it in more than one modules, so I've moved it to root object.

**Comment by akshayasalvi** [comment]
PR updated

File: src/CONST.js
Comment (Luke9389) [reply src/CONST.js]: Ok, I think that's reasonable 👍

**Review by Luke9389 — DISMISSED** [review]
Before merging I'd like to verify that you've tested these changes since the most recent commit. I ask all my contributors this question because it's easy to assume our commits don't break anything, and then they cause regressions. Lets get you paid on time :)

**Comment by akshayasalvi** [comment]
Yes tested :)

**Comment by akshayasalvi** [comment]
@Luke9389 @MonilBhavsar Anything that needs to be checked from my side here?

**Comment by Luke9389** [comment]
The only thing @akshayasalvi is that there's a merge conflict in this PR right now. Can you resolve that pls?

**Comment by akshayasalvi** [comment]
@Luke9389 Resolved and tested again

**Comment by Luke9389** [comment]
@akshayasalvi Looks like there's another conflict.

**Comment by akshayasalvi** [comment]
> @akshayasalvi Looks like there's another conflict.

Ohh no :(

**Comment by Luke9389** [comment]
Heh, it's OK! No worries, just be sure you're merging with the most recent version of our `main` branch. 
Usually I run `git fetch` to help with this.

**Comment by akshayasalvi** [comment]
Resolved again and tested

> Heh, it's OK! No worries, just be sure you're merging with the most recent version of our `main` branch. Usually, I run `git fetch` to help with this. (Though, if you want to be 200% sure, checkout main and run `git fetch && git pull`)

Yeah, I do it every time. It is just other changes are getting merged and resulting in conflicts. I am guessing these screens are under active development.

**Review by Luke9389 — APPROVED** [review]

**Comment by puneetlath** [comment]
Hm, that seems like a separate issue rather than a regression caused by this PR. This PR just enforces that what is entered is the right type and length. But doesn't do any validation on whether the numbers entered are valid. So I don't think we need to block this PR based on that.

**Comment by marcaaron** [comment]
Just noticed the regex initially added here allows people to enter phone numbers that are only 3 digits ?

Seems like something is wrong? Or is that expected..?

**Comment by marcaaron** [comment]
Yes, that's good. But what 3 digit number are they entering that we should accept as a phone number?

**Pull request merged**
